services/watson-extension/src/watson_extension/routes/platform/integrations.py

```diff
@@ -0,0 +1,322 @@
+    )
+
+    if integration_setup_success:
+        return RedhatIntegrationsSetupResponse(
+            response=await render_template(
+                "platform/integrations/redhat_integrations_setup_success.txt.jinja",
+            )
+        )
+
+    return RedhatIntegrationsSetupResponse(
+        response=await render_template(
+            "platform/integrations/integrations_setup_error.txt.jinja",
+        )
+    )
+
+
+@blueprint.get("/integrations_setup")
+@validate_querystring(IntegrationsSetupRequestQuery)
+@validate_response(IntegrationsSetupResponse)
+@document_headers(RHSessionIdHeader)
+async def integrations_setup(
+    query_args: IntegrationsSetupRequestQuery,
+    integrations_service: injector.Inject[IntegrationsCore],
+) -> IntegrationsSetupResponse:
+    if query_args.integration_type == IntegrationTypes.COMMUNICATIONS:
+        if await integrations_service.communications_integrations_setup(
+            setup_name=query_args.setup_name,
+            setup_url=query_args.setup_url,
+            setup_secret=query_args.setup_secret,
+            setup_use_secret=query_args.setup_use_secret,
+            setup_type=query_args.setup_type,
+        ):
+            return IntegrationsSetupResponse(
+                response=await render_template(
+                    "platform/integrations/communications_integrations_setup_success.txt.jinja",
+                    integration_setup_type=query_args.setup_type,
+                )
+            )
+
+    if query_args.integration_type == IntegrationTypes.REPORTING:
+        if await integrations_service.reporting_integrations_setup(
+            setup_name=query_args.setup_name,
+            setup_url=query_args.setup_url,
+            setup_secret=query_args.setup_secret,
+            setup_use_secret=query_args.setup_use_secret,
+            setup_type=query_args.setup_type,
+        ):
+            return IntegrationsSetupResponse(
+                response=await render_template(
+                    "platform/integrations/reporting_integrations_setup_success.txt.jinja",
+                    integration_setup_type=query_args.setup_type,
+                )
+            )
+
+    if query_args.integration_type == IntegrationTypes.WEBHOOK:
+        if await integrations_service.webhook_integrations_setup(
+            setup_name=query_args.setup_name,
+            setup_url=query_args.setup_url,
+            setup_secret=query_args.setup_secret,
+            setup_use_secret=query_args.setup_use_secret,
+        ):
+            return IntegrationsSetupResponse(
+                response=await render_template(
+                    "platform/integrations/webhook_integrations_setup_success.txt.jinja",
+                )
+            )
+
+    return IntegrationsSetupResponse(
+        response=await render_template(
+            "platform/integrations/integrations_setup_error.txt.jinja",
+        )
+    )
+
+
+@blueprint.get("/fetch_integrations/options")
```

Comment:
Not necessary, I still think following a standard format could be good, so that we end up with normal API routes, i.e.

/integrations/options
/integrations/setup
/integrations/redhat/setup
/integrations/actions
/integrations/update // might be a better way to organize that since it would be redundant with a post?

(integrations is already a part of the url from the blueprint too)